steam/game_servers.py

```diff
@@ -571,12 +576,12 @@ def a2s_rules(server_addr, timeout=2, challenge=0):
     rules = {}
 
     while len(rules) != num_rules:
-        name = data.read_cstring()
-        value = data.read_cstring()
+        name = data.read_cstring(binary=binary)
+        value = data.read_cstring(binary=binary)
 
-        if _re_match(r'^\-?[0-9]+$', value):
+        if not binary and _re_match(r'^\-?[0-9]+$', value):
             value = int(value)
-        elif _re_match(r'^\-?[0-9]+\.[0-9]+$', value):
+        elif not binary and _re_match(r'^\-?[0-9]+\.[0-9]+$', value):
```

Comment:
Should just be just one `if not binary:`

Reply:
Done